src/Form/widgets/SwitchWidget/SwitchWidget.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+
+import {IWidget} from '../../../../typings/IWidget';
+import Errors from '../../components/Errors';
+import validator from '../../Validator';
+import Switch from './components/Switch';
+
+interface TSwitchWidgetState {
+  value?: boolean | null | undefined;
+  errors: any[];
+}
+
+export class SwitchWidget extends React.Component<IWidget, TSwitchWidgetState> {
+  public static defaultProps = {
+    uiSchema: {},
+    isRequired: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      errors: [],
+      value: !!this.props.value || !!this.props.schema.default,
+    };
+  }
+
+  public get value(): boolean {
+    return this.state.value;
+  }
+
+  public get isValid(): boolean {
+    const {isRequired, schema} = this.props;
+    const errors = [];
+
+    if (isRequired && !this.value) {
```

Comment:
If field is required, this mean it should have value. In this statement it's will works wrong because field can have true or false value.